apps/factory/src/components/create_template.tsx

```diff
@@ -20,28 +20,43 @@ import { TemplateOutput as pt } from "~/validators/prompt_template";
 // import DeleteIcon from '@mui/icons-material/Delete';
 import AddCircleIcon from "@mui/icons-material/AddCircle";
 import { ModelTypeSchema } from "~/generated/prisma-client-zod.ts";
-
+import { ModelTypeType } from "~/generated/prisma-client-zod.ts";
 import { useForm } from "react-hook-form";
 import { zodResolver } from "@hookform/resolvers/zod";
 import type { CreateTemplateInput } from "~/validators/prompt_template";
 import { createTemplateInput } from "~/validators/prompt_template";
 import { FormTextInput } from "./form_components/formTextInput";
 import { FormDropDownInput } from "./form_components/formDropDownInput";
+import EditIcon from "@mui/icons-material/Edit";
+import { api } from "~/utils/api";
+import toast from "react-hot-toast";
 
 export function CreateTemplate({
   pp,
   onCreate,
   sx,
   status,
   customError,
+  ptId,
+  packageId,
 }: {
   pp: pp;
   onCreate: Function;
   sx?: any;
   status: string;
   customError: any;
+  ptId: string;
+  packageId: string;
 }) {
   const [isOpen, setIsOpen] = useState(false);
+  const [openEditTemplate, setOpenEditTemplate] = useState<boolean>(false);
+  const [defaultModelType, setDefaultModelType] = useState<
+    ModelTypeType | undefined
+  >("TEXT2TEXT");
```

Comment:
use literal from enum rather than redefining the string literal ModelTypeType.enum.TEXT2TEXT